src/impl/realm_coordinator.cpp

```diff
@@ -148,35 +147,52 @@ void RealmCoordinator::set_config(const Realm::Config& config)
 
 std::shared_ptr<Realm> RealmCoordinator::get_realm(Realm::Config config)
 {
-    std::lock_guard<std::mutex> lock(m_realm_mutex);
+    std::unique_lock<std::mutex> lock(m_realm_mutex);
 
     set_config(config);
 
+    auto schema = std::move(config.schema);
+    auto migration_function = std::move(config.migration_function);
```

Comment:
This was addressed by 204467252841de364d6ef8fc6ca36b773cba2a88, as the the migration function is now verified prior to this line.